content/learn/intermediate/outputs-2.md

```diff
@@ -0,0 +1,249 @@
+return a structured object that can be sent as a message, as demonstrated in the last line of the code.
+
+The `"title"` and `"description"` fields are self-explanatory---we can use Discord Markdown in the latter. The `"color"`
+field is in our case a hexadecimal color value that will be used as the color of the embed, but it can also take a
+[decimal value](https://www.binaryhexconverter.com/hex-to-decimal-converter).
+
+The `"fields"` field is a list (we call it slice) of dictionaries, where each dictionary represents a field in the embed.
+Each field dictionary must contain a `"name"` and a `"value"` field, and can optionally contain an `"inline"` field.
+This field is a boolean that determines whether the field should be displayed inline with the previous field.
+
+The `"author"` field is a dictionary that contains the name and icon URL of the author of the embed. In this case, we
+use the username of the user who triggered the custom command as the name, and the user's avatar URL as the icon URL.
+
+{{< callout context="tip" title="Tip" icon="outline/rocket" >}}
+
+There is a [community-made embed visualizer][v-embed] that can help you create custom command embeds more easily.
+However, it is still a good idea to not wholly rely on the GUI and understand the structure of the embeds yourself.
+
+For your convenience, we have [prefilled the above example][prefill] in the visualizer.
+
+[v-embed]: https://yagpdbembeds.netlify.app/
+[prefill]: https://yagpdbembeds.netlify.app/?data=JTdCJTIyZW1iZWQlMjIlM0ElN0IlMjJ0aXRsZSUyMiUzQSUyMlRoaXMlMjBpcyUyMGElMjB0aXRsZSUyMiUyQyUyMmRlc2NyaXB0aW9uJTIyJTNBJTIyVGhpcyUyMGlzJTIwYSUyMGRlc2NyaXB0aW9uLiUyMiUyQyUyMmNvbG9yJTIyJTNBMTY3MTE2ODAlMkMlMjJmaWVsZHMlMjIlM0ElNUIlN0IlMjJuYW1lJTIyJTNBJTIyRmllbGQlMjAxJTIyJTJDJTIydmFsdWUlMjIlM0ElMjJWYWx1ZSUyMDElMjIlMkMlMjJpbmxpbmUlMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMkZpZWxkJTIwMiUyMiUyQyUyMnZhbHVlJTIyJTNBJTIyVmFsdWUlMjAyJTIyJTJDJTIyaW5saW5lJTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMmF1dGhvciUyMiUzQSU3QiUyMm5hbWUlMjIlM0ElMjJsem9kZCUyMiUyQyUyMmljb25fdXJsJTIyJTNBJTIyaHR0cHMlM0ElMkYlMkZjZG4uZGlzY29yZGFwcC5jb20lMkZhdmF0YXJzJTJGNjY1MjQzNDQ5NDA1OTk3MDY2JTJGZDYzMTM3MWJiY2NhMDI3NDgxYTcxMjk1NGUyODQyYmMucG5nJTNGc2l6ZSUzRDI1NiUyMiU3RCU3RCU3RA==
+
+{{< /callout >}}
+
+## Editing Messages
+
+Sending a message is nice and all, but for the sake of keeping things clean, you might want to edit a message instead of
+creating a new one each time something changes. We provide the `editMessage` function for this purpose.
+
+```go
+{{ editMessage channel_id message_id new_message_content }}
+```
+
+The `channel_id` is the ID of the channel in which the message is located. The `message_id` is the ID of the message you
+want to edit. The `new_message_content` is the new content you want to set for the message, completely replacing it.
+You should note that YAGPDB can only edit messages from itself, just like you cannot edit someone else's messages.
```

Comment:
is the new content of the message, which will completely overwrite the existing content. See [editing-embeds] if you only want to change certain fields of an existing embed while leaving others intact.